src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.lib.Inject;
+import mate.academy.model.CinemaHall;
+import mate.academy.service.CinemaHallService;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+
+public class CinemaHallServiceImpl implements CinemaHallService {
+    @Inject
+    private CinemaHallDao cinemaHallDao;
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return cinemaHallDao.add(cinemaHall);
+    }
+
+    @Override
+    public CinemaHall get(Long id) {
+        return cinemaHallDao.get(id).get();
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        try {
+            return session.createQuery("From CinemaHall", CinemaHall.class).getResultList();
```

Comment:
The `getAll` method should delegate to the `cinemaHallDao.getAll()` method instead of directly using the Hibernate session. This ensures consistency and encapsulation of data access logic within the DAO layer.